eox_core/api/v1/serializers.py

```diff
@@ -79,6 +85,84 @@ def validate(self, attrs):
         return attrs
 
 
+class EdxappExtendedUserSerializer(EdxappUserSerializer):
+    """
+    This serializer allows admiting all the
+    UserProfile fields and also the extra fields for the
+    userProfile.meta values that are specified in the
+    microsite settings.
+    """
+    year_of_birth = serializers.ChoiceField(choices=YEAR_OF_BIRTH_CHOICES)
+    gender = serializers.ChoiceField(choices=get_gender_choices())
+    city = serializers.CharField()
+    goals = serializers.CharField()
+    bio = serializers.CharField(max_length=3000)
+    profile_image_uploaded_at = serializers.DateTimeField()
+    phone_number = serializers.CharField(max_length=50)
+    mailing_address = serializers.CharField()
+    courseware = serializers.CharField(max_length=255)
+    level_of_education = serializers.ChoiceField(choices=get_level_of_education_choices())
+    country = CountryField(required=False)
+
+    def __init__(self, *args, **kwargs):  # pylint: disable=too-many-locals
+        """
+        Add the custom registration fields specified in the settings.
+        """
+        super(EdxappExtendedUserSerializer, self).__init__(**kwargs)
+
+        extended_profile_fields = getattr(settings, 'extended_profile_fields', [])
+        extra_fields = getattr(settings, 'REGISTRATION_EXTRA_FIELDS', {})
+        ednx_custom_regitration_fields = getattr(settings, 'EDNX_CUSTOM_REGISTRATION_FIELDS', [])
+        copy_fields = self.fields
```

Comment:
Any reason to use `copy_fields`? if you pop from here you also pop from `self.fields`

Reply:
I changed this, now I'm just creating a list with all the names of the fields in it instead of using the object